reprounzip-singularity/Singularity_multiple_runs.py

```diff
@@ -0,0 +1,135 @@
+        for folder in folders:
+            os.mkdir(os.path.join(root_path,folder))
+        folders = ['actions','libs','env']
+        os.mkdir(SINGULARITY_DIR)
+        for folder in folders:
+            os.mkdir(os.path.join(SINGULARITY_DIR,folder)) 
+        singularoty_files = ["labels.json","runscript","startscript"]
+        for file in singularoty_files:
+            open(os.path.join(SINGULARITY_DIR,file), 'a').close()
+
+make_singularity_directories()
+
+def copy_action_files():
+    for file in os.listdir("../../singularitd_files/actions/"):
+        bashCommand = "cp ../../singularitd_files/actions/{} DATA/.singularity.d/actions/".format(file)
+        process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+        output, error = process.communicate()
+
+copy_action_files()
+
+def write_env_file(env,env_file):
+    env_file = os.path.join(ENV_DIR, env_file)
+    with open(env_file, 'w+') as f:
+        for key,value in env.items():
+            f.write(key+"='"+value+"'\n")
+    bashCommand = "chmod +x {}".format(env_file)
+    process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+    output, error = process.communicate()
+
+def make_environment_file():
+    env_files = [run_env_file,apps_file,base_file]
+    for file in env_files:
+        open(os.path.join(ENV_DIR,file), 'a').close()
+    #Write the environment file from config.yml file
+    source_config="METADATA/config.yml"
+    my_dict = yaml.load(open(source_config))
+    runs = my_dict.get('runs')
+    if len(runs) > 1:
+        # create one env file for each run:
+        for run in runs:
+            filename = run['id']+"_env.sh"
+            write_env_file(run.get('environ'), filename)
+    else:
+        print(len(runs))
+        write_env_file(runs[0].get('environ'),run_env_file)
+
+make_environment_file()
+
+def make_runscript():
+    source_config="METADATA/config.yml"
+    my_dict = yaml.load(open(source_config))
+    runs = my_dict['runs']
+    cmd="#!/bin/sh \n"
+    for run in runs:
+        binary = run['binary']
+        workingdir = run ['workingdir']
+        run_file = run['argv'][1]
+        if len(runs)>1:
+            cmd += "source /.singularity.d/env/{}_env.sh \n".format(run['id'])
+        cmd +="cd {0}\nexec {1} {2}\n".format(workingdir,binary,run_file)
+    with open(os.path.join(SINGULARITY_DIR, "runscript"), 'w') as f:
+        f.write(cmd)
+    bashCommand = "chmod +x {}/runscript".format(SINGULARITY_DIR)
+    process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+    output, error = process.communicate() 
+
+make_runscript()
+
+
+def copy_busybox():
+    if not os.path.exists(SHELL_DIR):
+    	os.makedirs(SHELL_DIR)
+    if not os.path.isfile(os.path.join(SHELL_DIR,"sh")):
+	print("no sh in bin")
+	bashCommand = "cp  ../bin/sh {}".format(SHELL_DIR)
+	process = subprocess.Popen(bashCommand.split(), stdout=subprocess.PIPE)
+    	output, error = process.communicate()
```

Comment:
You should probably always copy busybox, for example to `/busybox` (that's what reprounzip-docker does).

It can be that the `/bin/sh` in the package does not work, or does not support some things we need (like `exec -a` which we should use eventually)